src/main/java/ru/practicum/shareit/booking/model/Booking.java

```diff
@@ -1,17 +1,69 @@
 package ru.practicum.shareit.booking.model;
 
-import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.Hibernate;
 import ru.practicum.shareit.item.model.Item;
 import ru.practicum.shareit.user.model.User;
 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
 import java.time.LocalDateTime;
+import java.util.Objects;
 
-@Data
+@Entity
+@Table(name = "bookings")
+@Getter
+@Setter
+@ToString
+@RequiredArgsConstructor
+@AllArgsConstructor
 public class Booking {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
     private Long id;
-    private final LocalDateTime start;
-    private final LocalDateTime end;
-    private final Item item;
-    private final User booker;
+
+    @Column(name = "start_date")
+    private LocalDateTime start;
+
+    @Column(name = "end_date")
+    private LocalDateTime end;
+
+    @ManyToOne
+    @JoinColumn(name = "item_id")
+    private Item item;
+
+    @ManyToOne
+    @JoinColumn(name = "booker_id")
+    private User booker;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "status")
     private Status status;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
+        Booking booking = (Booking) o;
+        return id != null && Objects.equals(id, booking.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return getClass().hashCode();
```

Comment:
Расскажите, для чего это сделано ? И как вы это используете ? Равно как и метод equals

Reply:
По сути надо переопределить только `equals` для `User`, поскольку там равенство заключается в поле `email`, а не в `id`, как у остальных сущностей. Но идея подсвечивает аннотации `@Data` и `@EqualsAndHashCode` и настоятельно рекомендует сделать переопределения:`Using @Data for JPA entities is not recommended. It can cause severe performance and memory consumption issues.`

Оставлю переопределение только для `User`.

Reply:
Коротко, equals & hashCode не нужно переопределять. Даже в дипломе будет такое требование, чтобы не было реализаций этих методов в классах - сущностях. 

Там довольно много нюансов, в рамках данного контекста обучения даже программа вам этого не даст. 
Но вы можете почитать здесь https://thorben-janssen.com/ultimate-guide-to-implementing-equals-and-hashcode-with-hibernate/